contracts/StorageDeal.sol

```diff
@@ -0,0 +1,206 @@
+pragma solidity >=0.8.0;
+
+contract StorageDeals {
+    struct Deal {
+        address minerAddress;
+        address clientAddress;
+        uint256 startDate;
+        uint256 endDate;
+        uint256 pricePerByte;
+        uint256 dataSize;
+        uint256 status;
+        bytes32 fileHash;
+    }
+
+    mapping(bytes32 => Deal) public deals;
+
+    enum DealStatus {
+        Inactive,
+        Active,
+        Terminated
+    }
+
+    function createDeal(
+        address minerAddress,
+        address clientAddress,
+        uint256 startDate,
+        uint256 endDate,
+        uint256 pricePerByte,
+        uint256 dataSize,
+        bytes32 fileHash
+    ) public {
+        bytes32 dealId = sha256(
```

Comment:
how do we know the `dealId` in case I want to use the `queryDeal` or `updateDeal`?

Reply:
I think we will need to use js IPFS to get the CID from the deal id

https://js.ipfs.tech/